### PR TITLE
ISPN-5876 Invalidate keys on commit if 2-phase

### PR DESCRIPTION
**Pull request by galderz**
* Invalidate keys on commit if transaction is 2-phase, otherwise
  invalidate keys on prepare.

### CODE REVIEW

**Comment by elguardian** [comment]
BZ 6.4.z - https://bugzilla.redhat.com/show_bug.cgi?id=1273147
Jira: https://issues.jboss.org/browse/ISPN-5876

File: core/src/main/java/org/infinispan/util/logging/Log.java [hunk 1]
```diff
@@ -1371,4 +1371,12 @@ void preparedTxAlreadyExists(RecoveryAwareTransaction previous,
    @LogMessage(level = WARN)
    @Message(value = "The eviction max-entries attribute has been deprecated. Please use the size attribute instead", id = 383)
    void evictionMaxEntriesDeprecated();
+
+   @LogMessage(level = WARN)
+   @Message(value = "Unable to broadcast evicts as a part of the commit phase", id = 384)
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/util/logging/Log.java]: Shouldn't we call them invalidations?

**Comment by galderz** [comment]
@danberindei Fixed. 5.2.x branch has been fixed too.

**Comment by danberindei** [comment]
Integrated, thanks Galder!

**Pull request closed without merging**